modules/agent/pkg/deployer/normalizers/mutatingwebhook.go

```diff
@@ -0,0 +1,82 @@
+package normalizers
+
+import (
+    "github.com/sirupsen/logrus"
+    v1 "k8s.io/api/admissionregistration/v1"
+    "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+    "k8s.io/apimachinery/pkg/runtime"
+    "k8s.io/apimachinery/pkg/version"
+    "strings"
+)
+
+type MutatingWebhookNormalizer struct {
+    KubeVersion *version.Info
+}
+
+func (m MutatingWebhookNormalizer) Normalize(un *unstructured.Unstructured) error {
+    if un == nil {
+        return nil
+    }
+    gvk := un.GroupVersionKind()
+    if gvk.Group != "admissionregistration.k8s.io" || gvk.Kind != "MutatingWebhookConfiguration" {
+        return nil
+    }
+
+    var webhook v1.MutatingWebhookConfiguration
+    err := runtime.DefaultUnstructuredConverter.FromUnstructured(un.Object, &webhook)
+    if err != nil {
+        logrus.Error("Failed to convert unstructured to webhook")
+        return nil
+    }
+
+    for i, config := range webhook.Webhooks {
+        if strings.HasSuffix(string(config.ClientConfig.CABundle), "\n") {
+            config.ClientConfig.CABundle = nil
+        }
+        if config.SideEffects == nil {
+            if gvk.Version == "v1beta1" {
+                unknown := v1.SideEffectClassUnknown
+                config.SideEffects = &unknown
+            }
+        }
+        if config.TimeoutSeconds == nil {
+            switch m.KubeVersion.Minor {
+            case "19":
+                config.TimeoutSeconds = &[]int32{30}[0]
+            case "20":
+                config.TimeoutSeconds = &[]int32{10}[0]
```

Comment:
Do we need a default case?

Reply:
this is the wrong implemtation and I have changed